.github/workflows/openssl.yml

```diff
@@ -61,6 +58,51 @@ jobs:
           cd openssl
           ./config --prefix=/opt && make && make install_sw
 
+  build:
+    name: Build and test pkcs11-provider
+    needs: setup
+    runs-on: ubuntu-22.04
+    container: fedora:latest
+    steps:
+      - name: Get Date for DNF cache entry
+        id: get-date-dnf
+        run: |
+          echo "date=$(/bin/date -u "+%Y%V")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: DNF cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            /var/cache/libdnf5
+          key: ${{ runner.os }}-dnf-openssl-${{ steps.get-date-dnf.outputs.date }}
+
+      - name: Install Dependencies
+        run: |
+            dnf -y install clang meson cargo expect pkgconf-pkg-config opensc \
+              p11-kit-devel gcc g++ sqlite-devel python3-six which cmake \
+              nss-softokn nss-tools nss-softokn-devel nss-devel softhsm
+
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Get Date for /opt cache entry
+        id: get-date-opt
+        run: |
+          echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: /opt cache
+        id: cache-opt
+        uses: actions/cache@v4
+        with:
```

Comment:
I think you should add the option `fail-on-cache-miss: true` here so that we do not get confused about why the build fails if the openssl cache somehow failed to be retrieved

I would also name the steos something explicitly like "Fetch openssl build from cache"

Reply:
It's there - just a little below (line 102) but fail-on-cache-miss will be a better way, I'll fix that (and name as well).

Reply:
Fixed.